src/robot/libraries/BuiltIn.py

```diff
@@ -1662,26 +1662,58 @@ def run_keyword_and_expect_error(self, expected_error, name, *args):
         return unic(error)
 
     @run_keyword_variant(resolve=2)
-    def repeat_keyword(self, times, name, *args):
+    def repeat_keyword(self, repeat, name, *args):
         """Executes the specified keyword multiple times.
 
         ``name`` and ``args`` define the keyword that is executed
-        similarly as with `Run Keyword`, and ``times`` specifies how many
-        the keyword should be executed. ``times`` can be given as an
+        similarly as with `Run Keyword`.
+
+        If ``repeat`` is given as count, it specifies how many the keyword
+        should be executed. ``repeat`` can be given as an
         integer or as a string that can be converted to an integer. If it is
         a string, it can have postfix ``times`` or ``x`` (case and space
         insensitive) to make the expression more explicit.
 
-        If ``times`` is zero or negative, the keyword is not executed at
+        If ``repeat`` is given as timeout, it must be in Robot Framework's
+        time format (e.g. ``1 minute``, ``2 min 3 s``) except a number like
+        42 is interpreted as count, so that a floating point number like
+        4.2 would raise exception.
```

Comment:
This sounds a bit complicated. Perhaps something like this would be better:

> If ``repeat`` is given as timeout, it must be in Robot Framework's  time format (e.g. ``1 minute``, ``2 min 3 s``). Using a number alone (e.g. `1` or `1.5`) does not work in this context.